cloudformation/Prowler-Template.yaml

```diff
@@ -679,6 +680,85 @@ Resources:
           TRACKING_TABLE_NAME: !Ref ProwlerFindingsTrackingTable
           OUTPUT_BUCKET: !Ref ProwlerBucket
 
+  FindingsAlertRole:
+    Type: AWS::IAM::Role
+    Condition: cDeployAlertLambda
+    Properties:
+      AssumeRolePolicyDocument:
+        Version: '2012-10-17'
+        Statement:
+        - Effect: Allow
+          Principal:
+            Service:
+            - lambda.amazonaws.com
+          Action:
+          - sts:AssumeRole
+      Path: /
+      Policies:
+      - PolicyName: LambdaLogging
+        PolicyDocument:
+          Version: '2012-10-17'
+          Statement:
+          - Resource: '*'
+            Action:
+            - logs:*
+            Effect: Allow
+      - PolicyName: DDBStreamRead
+        PolicyDocument:
+          Version: '2012-10-17'
+          Statement:
+            - Effect: Allow
+              Action:
+                - dynamodb:DescribeStream
+                - dynamodb:GetRecords
+                - dynamodb:GetShardIterator
+                - dynamodb:ListStreams
+              Resource:
+                - !GetAtt ProwlerFindingsTrackingTable.StreamArn
+      - PolicyName: AccessSlackSecret
+        PolicyDocument:
+          Statement:
+          - Effect: Allow
+            Action:
+              - ssm:GetParameters
+              - secretsmanager:GetSecretValue
+              - kms:Decrypt
+            Resource: !Sub "arn:aws:secretsmanager:${AWS::Region}:${AWS::AccountId}:secret:${pSlackSecret}*"
+      - PolicyName: S3ConfigFileAccess
+        PolicyDocument:
+          Version: '2012-10-17'
+          Statement:
+          - Action:
+            - s3:GetObject
+            Effect: Allow
+            Resource:
+              - !Sub 'arn:aws:s3:::${pBucketName}/*.yaml'
+
+  FindingsAlertFunction:
+    Type: AWS::Serverless::Function
+    Condition: cDeployAlertLambda
+    Properties:
+      FunctionName: !Sub "${AWS::StackName}-findings-alert"
+      Description: Send alerts about new findings
+      Handler: alert_prowler_findings.handler
+      Role: !GetAtt FindingsAlertRole.Arn
+      CodeUri: ../lambda
+      Timeout: 60
+      Environment:
+        Variables:
+          SLACK_SECRET: !Sub "arn:aws:secretsmanager:${AWS::Region}:${AWS::AccountId}:secret:${pSlackSecret}"
+          CONFIG_BUCKET: !Ref ProwlerBucket
+      Layers:
+        - arn:aws:lambda:us-east-1:177933569100:layer:AWS-Parameters-and-Secrets-Lambda-Extension:11
```

Comment:
What is this layer?

Reply:
basically in-memory caching for secrets. reduces calls to SM - faster and cheaper. It is an [AWS-provided layer](https://docs.aws.amazon.com/systems-manager/latest/userguide/ps-integration-lambda-extensions.html#ps-integration-lambda-extensions-add). [Blog post](https://aws.amazon.com/blogs/compute/using-the-aws-parameter-and-secrets-lambda-extension-to-cache-parameters-and-secrets/) and [doc](https://docs.aws.amazon.com/secretsmanager/latest/userguide/retrieving-secrets_lambda.html) describing how it works.

Reply:
couple other use examples https://github.com/aws-samples/serverless-patterns/blob/8d77bb0725e257c32bb84b8e1ead9b0bd2ff600b/lambda-secrets-manager-extension-python/hello_world/app.py#L10 and https://github.com/jfrog/artifactory-sagemaker/blob/main/infrastructure/lambda_function.py

Reply:
Makes total sense, and this is a nifty solution. 

But, in this  case we're pulling just the Slack Secret, which will rarely change. This probably doesn't need the performance boost of the caching. 

Rather I'd be concerned around the usage of an unknown layer in my client's environment, and the same questions I asked you being asked of me.